hail_search/queries/mito.py

```diff
@@ -112,6 +118,215 @@ class MitoHailTableQuery(BaseHailTableQuery):
+        wes_passes = ht[WES_FAMILY_ENTRIES_FIELD] if wes_passes_quality_filter is None else (
+            ht[WES_FAMILY_ENTRIES_FIELD].map(lambda entries: hl.or_missing(wes_passes_quality_filter(entries), entries))
+        )
+        ht = ht.annotate(wes_passes_quality=wes_passes)
+
+        wgs_passes_quality_filter = self._get_family_passes_quality_filter(quality_filter, ht, WGS_FILTERS_FIELD, **kwargs)
+        wgs_passes = ht[WGS_FAMILY_ENTRIES_FIELD] if wgs_passes_quality_filter is None else (
+            ht[WGS_FAMILY_ENTRIES_FIELD].map(lambda entries: hl.or_missing(wgs_passes_quality_filter(entries), entries))
+        )
+        ht = ht.annotate(wgs_passes_quality=wgs_passes)
+        return ht
+
+    @staticmethod
+    def _apply_filters(ht, family_idx_map) -> hl.Table:
+        # Keep family from both sample types if either passes quality and either passes inheritance
+        ht = ht.annotate(
+            wes_family_entries=hl.enumerate(ht[WES_FAMILY_ENTRIES_FIELD]).map(
+                lambda x: hl.or_missing(  # x[0] is family index, x[1] is list of samples
+                    (hl.is_defined(ht.wes_passes_quality[x[0]]) | hl.if_else(
+                        family_idx_map.get(x[1][0]['familyGuid'], hl.empty_dict(hl.tstr, hl.tint32)).contains(SAMPLE_TYPE_WGS),
+                        hl.is_defined(ht.wgs_passes_quality[family_idx_map[x[1][0]['familyGuid']][SAMPLE_TYPE_WGS]]),
+                        False
+                    )) & (hl.is_defined(ht.wes_passes_inheritance[x[0]]) | hl.if_else(
+                        family_idx_map.get(x[1][0]['familyGuid'], hl.empty_dict(hl.tstr, hl.tint32)).contains(SAMPLE_TYPE_WGS),
+                        hl.is_defined(ht.wgs_passes_inheritance[family_idx_map[x[1][0]['familyGuid']][SAMPLE_TYPE_WGS]]),
+                        False
+                    )), x[1]
+            )),
+            wgs_family_entries=hl.enumerate(ht[WGS_FAMILY_ENTRIES_FIELD]).map(
+                lambda x: hl.or_missing(
+                    (hl.is_defined(ht.wgs_passes_quality[x[0]]) | hl.if_else(
+                        family_idx_map.get(x[1][0]['familyGuid'], hl.empty_dict(hl.tstr, hl.tint32)).contains(SAMPLE_TYPE_WES),
+                        hl.is_defined(ht.wes_passes_quality[family_idx_map[x[1][0]['familyGuid']][SAMPLE_TYPE_WES]]),
+                        False
+                    )) & ( hl.is_defined(ht.wgs_passes_inheritance[x[0]]) | hl.if_else(
+                        family_idx_map.get(x[1][0]['familyGuid'], hl.empty_dict(hl.tstr, hl.tint32)).contains(SAMPLE_TYPE_WES),
+                        hl.is_defined(ht.wes_passes_inheritance[family_idx_map[x[1][0]['familyGuid']][SAMPLE_TYPE_WES]]),
+                        False
+                    )), x[1]
+            ))
+        )
+        ht = ht.transmute(
+            family_entries=hl.coalesce(
+                ht.wes_family_entries, hl.empty_array(ht.wes_family_entries.dtype.element_type)).extend(
```

Comment:
Is there a reason you can't just do `ht.wes_family_entries.extend(ht.wgs_family_entries)`

Reply:
![image](https://github.com/user-attachments/assets/b8374429-826c-4e78-b89c-fedc0c39e270)
If that family doesn't have an entry at all for a variant in ht, it is NA after I join the two tables together.